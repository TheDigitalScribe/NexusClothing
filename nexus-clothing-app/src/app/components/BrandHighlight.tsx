import React from 'react'

interface BrandHighlight {
  Icon: React.ComponentType<React.SVGProps<SVGSVGElement>>;
  title: string;
  subtext: string;
}

export const BrandHighlight = ({ Icon, title, subtext }: BrandHighlight) => {
  return (
    <div className="flex flex-col items-center w-60">
      <Icon className="w-12 h-12 text-blue-500" />
      <h2 className="text-lg font-bold mt-2">{title}</h2>
      <p className="leading-1 mt-2">{subtext}</p>
    </div>
  )
}
