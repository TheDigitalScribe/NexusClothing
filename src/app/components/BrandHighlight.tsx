import React from 'react'
import { BrandHighlightProps } from '../types'

export const BrandHighlight = ({ Icon, title, subtext }: BrandHighlightProps) => {
  return (
    <div className="flex flex-col items-center w-60">
      <Icon className="w-12 h-12 text-blue-500" />
      <h3 className="text-lg font-bold mt-2">{title}</h3>
      <p className="leading-relaxed mt-2">{subtext}</p>
    </div>
  )
}
