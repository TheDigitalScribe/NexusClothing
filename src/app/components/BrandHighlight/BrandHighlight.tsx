import React from 'react'
import { BrandHighlightProps } from '../../../types/types'

export const BrandHighlight = ({ Icon, title, subtext }: BrandHighlightProps) => {
  return (
    <div className="flex flex-col items-center w-60">
      <Icon data-testid="brand-highlight-icon" className="w-12 h-12 text-blue-500" />
      <h3 className="text-md lg:text-xl font-bold mt-2">{title}</h3>
      <p className="text-sm text-center lg:text-md leading-relaxed mt-2">{subtext}</p>
    </div>
  )
}
