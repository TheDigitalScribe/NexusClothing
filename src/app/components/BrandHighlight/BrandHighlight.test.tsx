import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrandHighlight } from '@/app/components/BrandHighlight/BrandHighlight';
import { FiStar, FiHome } from 'react-icons/fi';

const mockProps = {
  Icon: FiStar,
  title: 'Test Title',
  subtext: 'Test Subtext',
}

const mockPropsLongText = {
  Icon: FiStar,
  title: 'This is a very long title that might wrap',
  subtext: 'This is a very long subtext that might wrap to multiple lines and should still look good',
}

describe('BrandHighlight', () => {
  // Render test
  it('renders the component', () => {
    render(<BrandHighlight {...mockProps} />);

    expect(screen.getByText(mockProps.title)).toBeInTheDocument();
    expect(screen.getByText(mockProps.subtext)).toBeInTheDocument();
    expect(document.querySelector('svg')).toBeInTheDocument();
  });

  // Tailwind CSS classes test
  it('applies correct Tailwind CSS classes', () => {
    render(<BrandHighlight {...mockProps} />);

    expect(screen.getByText(mockProps.title)).toHaveClass('text-lg sm:text-lg lg:text-xl font-bold mt-2');
    expect(screen.getByText(mockProps.subtext)).toHaveClass('leading-relaxed mt-2');
    expect(document.querySelector('svg')).toHaveClass('w-12 h-12 text-blue-500');
  });

  // Different icons test
  it('works with different icons', () => {
    render(<BrandHighlight {...mockProps} Icon={FiStar} />);

    expect(document.querySelector('svg')).toBeInTheDocument();

    render(<BrandHighlight {...mockProps} Icon={FiHome} />);

    expect(document.querySelector('svg')).toBeInTheDocument();
  });

  // Long content test
  it('handles long text content', () => {
    render(<BrandHighlight {...mockPropsLongText} />);

    expect(screen.getByText(mockPropsLongText.title)).toBeInTheDocument();
    expect(screen.getByText(mockPropsLongText.subtext)).toBeInTheDocument();
  });

  // Snapshot test
  it('matches snapshot', () => {
    const { container } = render(<BrandHighlight {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});

