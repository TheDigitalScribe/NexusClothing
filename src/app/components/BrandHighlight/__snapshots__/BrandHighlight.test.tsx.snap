// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandHighlight matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col items-center w-60"
  >
    <svg
      class="w-12 h-12 text-blue-500"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
      />
    </svg>
    <h3
      class="text-lg sm:text-lg lg:text-xl font-bold mt-2"
    >
      Test Title
    </h3>
    <p
      class="leading-relaxed mt-2"
    >
      Test Subtext
    </p>
  </div>
</div>
`;
